### PR TITLE
add a subcommand for the connmgr

### PR DESCRIPTION
**Pull request by whyrusleeping**
License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
(note: this relies on uncommitted changes to the connmgr lib)

File: core/commands/swarm.go [hunk 1]
```diff
@@ -812,3 +814,72 @@ func filtersRemove(r repo.Repo, cfg *config.Config, toRemoveFilters []string) ([
...
+		if n.PeerHost == nil {
+			res.SetError(errNotOnline, cmds.ErrClient)
+			return
+		}
+
+		infos := make(map[string]interface{})
+		cmgr := n.PeerHost.ConnManager()
+		switch cmgr := cmgr.(type) {
+		case nil:
+			infos["type"] = "<nil>"
```
Comment (Stebalien) [inline core/commands/swarm.go]: Is distinguishing between these two cases really useful? Users aren't going to know what `<nil>` means.

File: core/commands/swarm.go [hunk 2]
```diff
@@ -812,3 +814,72 @@ func filtersRemove(r repo.Repo, cfg *config.Config, toRemoveFilters []string) ([
...
+		case *ifconnmgr.NullConnMgr:
+			infos["type"] = "disabled"
+		case *connmgr.BasicConnMgr:
+			infos["type"] = "basic"
+
+			inf := cmgr.GetInfo()
+			infos["lowWater"] = inf.LowWater
```
Comment (Stebalien) [inline core/commands/swarm.go]: I assume these are supposed to be human readable keys? If so, we should probably just say things like "Low Watermark" instead.

File: core/commands/swarm.go (hunk 1 above)
Comment (whyrusleeping) [reply core/commands/swarm.go]: They shouldnt ever see "nil", Everything currently will either return a NullConnMgr or a BasicConnMgr. So maybe I should remove it?

Comment (Stebalien) [reply core/commands/swarm.go]: Probably. I'd fail with an error if this isn't supposed to happen.

File: core/commands/swarm.go (hunk 2 above)
Comment (Kubuxu) [reply core/commands/swarm.go]: Those will be the result of JSON endpoint. We can create mappings in TextMarshaller.

**Comment by Kubuxu** [comment]
@whyrusleeping so how about making human readable format having really human readable keys (`lowWater => Low Watermark` and so on)?

**Comment by whyrusleeping** [comment]
@GitCop what is your problem? I have that.

**Comment by whyrusleeping** [comment]
Can someone take this over for me?

**Comment by lidel** [comment]
ConnMgr work continued in https://github.com/ipfs/kubo/issues/9420

**Pull request closed without merging**